feat: add support of heightmap in ground geometry inspector

julien-moreau opened the pull request.

yuripourre commented:
@julien-moreau it is not working for me, when I assign a height map it fails with error:

```
[.WebGL-0x2604062dea00] GL_INVALID_OPERATION: Insufficient buffer size.
```

yuripourre commented:
I cannot approve it but the code is good. I have only one comment.

julien-moreau commented:
> @julien-moreau ~it is not working for me, when I assign a height map it fails with error:~
> 
> ```
> [.WebGL-0x2604062dea00] GL_INVALID_OPERATION: Insufficient buffer size.
> ```
> 
> By bad, somehow I managed to set 0 subdivisions.
> 
> I could test it, it actually is more performant than my branch, it updates very smoothly.

Ok I found the problem. By default subdivisions is equal to 0 for the ground in the templates :(
I propose this at the moment right before generating from heightmap.

```typescript
if (subdivisions <= 1) {
	this.props.object.metadata.subdivisions = 32;
}
```

yuripourre commented on editor/src/editor/layout/inspector/mesh/geometry.tsx:
```diff
@@ -88,26 +106,100 @@ export class MeshGeometryInspector extends Component<IMeshGeometryInspectorProps
+		this.props.object.metadata.smoothFactor ??= 0;
+		this.props.object.metadata.heightMapTexturePath ??= null;
+		this.props.object.metadata.alphaFilter ??= 0;
+		this.props.object.metadata.colorFilter ??= [0.3, 0.59, 0.11];
```
@julien-moreau I always set the value to 1,1,1. 😅 

I think the greyscale values are useful for colored images but I always use black and white images. So for me is always an extra step. But if you or other users use colored images, I can live with that.

julien-moreau replied:
Let's go! :)

julien-moreau commented:
> Yeah, I think is better than not having any feedback at all. Maybe you could display a warning in the toast if it's no too much work. Or a message in the console.

That was a really good idea so I added both (toast that tells subdivisions changed, more details in console).
I also replaced the default color filter.

I'm working a bit the design and we are good to go :)
